src/__tests__/element-queries.js

```diff
@@ -193,4 +193,30 @@ test('using jest helpers to check element class names', () => {
   ).toThrowError()
 })
 
+test('test the debug helper prints the dom state here', () => {
+  const Large = `<div>
+        ${Array.from({length: 7000}, (v, key) => key).map(() => {
+          return `<div data-testid="debugging" data-otherid="debugging">
+                Hello World!
+            </div>`
+        })}
+    </div>`
+
+  const {getByText} = render(Large) // render large DOM which exceeds 7000 limit
+  expect(() => expect(getByText('not present')).toBeInTheDOM()).toThrowError()
+
+  const Hello = `<div data-testid="debugging" data-otherid="debugging">
+        Hello World!
+    </div>`
+  const {getByTestId} = render(Hello)
+  process.env.DEBUG_PRINT_LIMIT = 10 // user should see `. . .`
+  expect(() => expect(getByTestId('not present')).toBeInTheDOM()).toThrowError()
+
+  const {getByLabelText} = render(Hello)
+  process.env.DEBUG_PRINT_LIMIT = 10000 // user shouldn't see `. . .`
+  expect(() =>
+    expect(getByLabelText('not present')).toBeInTheDOM(),
+  ).toThrowError()
```

Comment:
Make sure to reset `DEBUG_PRINT_LIMIT` to clean things up.

It should look something like this:

```javascript
test('test the debug helper prints the dom state here', () => {
  const originalDebugPrintLimit = process.env.DEBUG_PRINT_LIMIT

  // the test contents

  process.env.DEBUG_PRINT_LIMIT = originalDebugPrintLimit
})
```

That way it doesn't mess up other tests.